common/dev/snoop.c

```diff
@@ -120,10 +110,21 @@ void snoop_read()
 		LOG_ERR("snoop read buffer alloc fail");
 		return;
 	}
-	rc = snoop_aspeed_register_rx_callback(snoop_dev, snoop_rx_callback);
-	if (rc != 0) {
-		printk("Cannot register RX callback\n");
-		return;
+
+	while (1) {
+		rc = snoop_aspeed_read(snoop_dev, 0, snoop_data, true);
+		if (rc == 0) {
+			proc_postcode_ok = true;
+			if (!k_mutex_lock(&snoop_mutex, K_MSEC(1000))) {
+				snoop_read_buffer[snoop_read_num % SNOOP_MAX_LEN] = *snoop_data;
+				snoop_read_num++;
+			} else {
+				LOG_ERR("snoop read lock fail");
+			}
+			if (k_mutex_unlock(&snoop_mutex)) {
```

Comment:
You will still try to unlock if mutex lock fail. If lock failed, unlocked will fail too. Better unlock in case of success.

Reply:
Hi @jagpalgill ,
I have modifed the code base on your suggestion. Thanks.
Please help review again.

Best Regards,
Sean